_posts/pt_BR/2016-01-01-angularjs-digest-vs-apply.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: post
+
+title: AngularJs - `$digest` vs `$apply`
+tip-number: 01
+tip-username: loverajoel 
+tip-username-profile: https://github.com/loverajoel
+tip-tldr: JavaScript modules e build steps estão ficando numerosos e complicados, mas e boilerplate e novos frameworks?
+
+categories:
+    - pt_br
+---
+
+Uma das características mais apreciadas do AngularJs é o two-way data binding. Para fazer este trabalho o AngularJs avalia as mudanças entre o model e a view através de ciclos ( $digest ). Você precisa entender esse conceito para entender como o framework funciona debaixo do capô.
+
+Angular avalia cada observador quando um evento é disparado. Esse é o ciclo conhecido como `$digest`. Algumas vezes você precisa forçar a execução de um novo ciclo manualmente e deve escolher a opção correta pois esta fase é uma das que mais influencia em termos de performance.
+
+### `$apply`
+Este método do core permite você iniciar o ciclo digestion explicitamente. Isso significa que todos os observadores são verificados; toda a aplicação inicia o `$digest loop`. Internamente, depois de executar um parâmetro de função opcional, ele chama `$rootScope.$digest();`.
```

Comment:
You can translate digestion cycle either as "ciclo de digestão" or better "ciclo digest" because it is understood inside the context.